lib/devise/encryptable/encryptors/pbkdf2.rb

```diff
@@ -0,0 +1,25 @@
+begin
+  module Devise
+    module Encryptable
+      module Encryptors
+        class Pbkdf2 < Base
+          def self.compare(encrypted_password, password, stretches, salt, pepper)
+            value_to_test = self.digest(password, stretches, salt, pepper)
+            ActiveSupport::SecurityUtils.fixed_length_secure_compare(encrypted_password, value_to_test)
+          end
+
+          def self.digest(password, stretches, salt, pepper)
+            hash = OpenSSL::Digest::SHA512.new
+            OpenSSL::KDF.pbkdf2_hmac(
+              password,
```

Comment:
If a password is entirely numeric this function will fail trying due to implicit conversion of integer to string. 

```suggestion
              password.to_s,
```